### PR TITLE
[#4901]: Fix npm version issue; updated scripts/setup

### LINKED ISSUE #4901: Fix npm version issue
### User Story

As a developer, I want a reliable build process

### Description

`npm install` stales, sometimes.
Find out the root cause and fix it

*Type*: Bug

### PR DESCRIPTION
**Pull request by alwx**
Fixes #4901 

Tested on **node v7.10.1 (npm v5.5.1)** and **node v9.11.2 (npm v6.1.0)**.
Node 10 is not supported by Realm: https://github.com/realm/realm-js/issues/1813

Also updated our build scripts, added the installation of Clojure CLI there (to build status with `clj-rn`) and made sure it works.

### CODE REVIEW

File: ios/Podfile.lock [hunk 1]
```diff
@@ -16,30 +16,44 @@ PODS:
...
+    - yoga (= 0.55.4.React)
+  - RNKeychain (3.0.0-rc.3):
+    - React
+  - yoga (0.55.4.React)
 
 DEPENDENCIES:
   - FirebaseMessaging
   - React (from `../node_modules/react-native`)
   - react-native-background-timer (from `../node_modules/react-native-background-timer`)
+  - RNKeychain (from `../node_modules/react-native-keychain`)
```
Comment (jeluard) [inline ios/Podfile.lock]: Is this related to the PR?

File: ios/Podfile.lock [hunk 2]
```diff
@@ -48,10 +62,11 @@ SPEC CHECKSUMS:
...
+  React: aa2040dbb6f317b95314968021bd2888816e03d5
+  react-native-background-timer: 63dcbf37dbcf294b5c6c071afcdc661fa06a7594
+  RNKeychain: 627c6095cef215dd3d9804a9a9cf45ab96aa3997
+  yoga: a23273df0088bf7f2bb7e5d7b00044ea57a2a54a
 
-PODFILE CHECKSUM: 653ad2d673bf8aa8127325a078e07aedc8dbe463
+PODFILE CHECKSUM: 58fa7be75df3ece53c570dbd30d721f4b6d77137
 
-COCOAPODS: 1.3.1
+COCOAPODS: 1.5.3
```
Comment (jeluard) [inline ios/Podfile.lock]: I'd rather leave pod changes out if they are not strictly mandatory.

**Review by jeluard — APPROVED** [review]

**Review by dmitryn — APPROVED** [review]
Just to confirm, there are a lot of version changes in `package-lock.json`, is it all required for this upgrade?

File: ios/Podfile.lock (hunk 1 above)
Comment (cammellos) [reply ios/Podfile.lock]: I get the same changes automatically in Xcode in a different PR https://github.com/status-im/status-react/pull/4988/files , probably at some point some code got through which did not update these files, so it's probably safer to include them at some point, otherwise we might be using different versions locally from prod (0.53 vs 0.55 seems like a big difference), but I know very little of mac build
